### ISSUE TITLE
Missing "EventSet" implentation / Feature Request

### ISSUE THREAD

**Issue by diclophis** [newrelic/infra-integrations-sdk#32]
It would appear based on the help docs: https://docs.newrelic.com/docs/infrastructure/integrations-sdk/file-specifications/integration-executable-file-specifications#event-data that integration binarys should be able to emit "Event Set" data to allow for tracking of "one-off messages for key activities on a system" ... However I do not see any code in this SDK that would facilitate / exemplify how the format of that JSON might be.

##### sdk version

- [ ] dev
- [x] 1.0
- [ ] ...

##### Golang version

- [x] All of them
- [ ] 1.8
- [ ] 1.7
- [ ] ...

##### OS

- [x] All of them
- [ ] Amazon Linux, all versions
- [ ] CentOS, version 6 or higher
- [ ] Debian, version 7 ("Wheezy") or higher
- [ ] Red Hat Enterprise Linux (RHEL), version 6 or higher
- [ ] Ubuntu, versions 12.04, 14.04, and 16.04 (LTS  versions)
- [ ] Windows Server, 2008 and 2012 and their service packs

**Comment by ikavanaghNR** [comment]
Hi,

Sorry for the long delay in getting back to you and thanks for pointing this out to us. We will add some info to make it more clear.  For now you can use the following to read in event data.

integration.Events = append(integration.Events, sdk.Event{"summary": "event message"})

Ian

**Comment by diclophis** [comment]
I have found the docs on how to make a executable that emits the data, so i am not currently blocked on this library supporting this feature... but it does seem like it would be a nice-to-have

**Comment by varas** [comment]
Hi @diclophis it has been added on the v3, so you now have 2 event constructors:

https://github.com/newrelic/infra-integrations-sdk/blob/v3.0.2/data/event/event.go#L15-L26

This should work for all the OSs and Golang versions.

There is no plan to keep updating old SDK versions, just bug and security fixed will be targeted on v1 and v2.

**Comment by diclophis** [comment]
cool thanks
